### DISCUSSION TITLE
python 如何用G2图表 使用html 以及jupyter上？

### DISCUSSION THREAD

**Discussion by usekevinwu** [antvis/G2#5570] [Ideas]
python 如何用G2以及在jupyter上使用，需要导入html ，然后用jupyter 去实现读取html ?

**Comment by hustcc** [comment]
需要额外封装一下就可与在 jupyter 中使用了，可以参考 [PyG2Plot](https://github.com/hustcc/PyG2Plot)，封装之后，我们可以在官网收录进来。

**Reply by usekevinwu** [reply]
谢谢您的答复，
看到pyg2plot是您这边实现的，不知道使用核心三部分代码模块：engine，plot,以及templates进行实现？
代码量如果修改比较大，我可能独立实现不了，主要是对前端一些还不熟悉。

**Reply by hustcc** [reply]
改动应该不大，原理都是使用 python 生成 js 的 G2 模板代码，你可以建一个 repo，关于前端部分，我可以给一些指导。主要是我这里没有 python 的环境，然后对于 python 发包之类的搞得不太懂。

**Reply by usekevinwu** [reply]
我也不没认真看过一些发包的问题，不过现在问题是我看到在pyg2plot 会在plot 方法里加了一个set_option，那么G2是否需要加这个配置项？

**Reply by hustcc** [reply]
g2 也有这个 chart.options(spec) 的方法。

**Reply by usekevinwu** [reply]
谢谢，我先尝试下，我看到G2(spec)有个option，不知道里面G2以同样的方法设置set_option 是否可行
![image](https://github.com/antvis/G2/assets/37398179/a62202f8-f94d-4f04-b897-627f2aa5bb18)

**Reply by usekevinwu** [reply]
以下都是基于pyg2plot再修改实现的
![image](https://github.com/antvis/G2/assets/37398179/71445913-d6be-4a7a-810f-29901c427b79)
![image](https://github.com/antvis/G2/assets/37398179/3e172401-7646-4b1b-99f3-2f072d503489)

options 测试了好几次，好像是引用mini.js存在小问题不太懂。

换了方向用G2的JS代码实现。

**Reply by hustcc** [reply]
必须要要改成使用 g2.min.js 文件啊~

**Reply by Baili-BL** [reply]
这个问题，在前面解决了。在set_options 里面加了一个js_code.前面不是mini.js的问题，
![image](https://github.com/antvis/G2/assets/35422348/e4eff597-af78-494c-983d-78d3e67f56cc)

但现在的问题是jupyter_notebook中不展示，但是在jupytert_lab中是展示的，貌似jupyter_lab兼容更好点。应该是templates中notebook需要改下,但是还是不知道怎么来改，应该是jupyter_notebook版本的问题。
